File: eCommerce/eCommerce.Services/Database/Notifikacija.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace eCommerce.Services.Database
{
    public class Notifikacija
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int KorisnikId { get; set; }

        [ForeignKey("KorisnikId")]
        public User Korisnik { get; set; } = null!;

        [Required]
        [MaxLength(150)]
        public string Naslov { get; set; } = string.Empty;

        [Required]
        [MaxLength(1000)]
        public string Poruka { get; set; } = string.Empty;

        [Required]
        public NotifikacijaTip Tip { get; set; } = NotifikacijaTip.Opsta;

        public bool Procitano { get; set; } = false;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public int? TerminId { get; set; }

        [ForeignKey("TerminId")]
        public Termin? Termin { get; set; }
    }

    public enum NotifikacijaTip
    {
        PodsjetnikTermina,
        TerminPotvrdjen,
        TerminOtkazan,
        PlacanjeUspjesno,
        PlacanjeVraceno,
        Opsta
    }
}

File: eCommerce/UI/ecommerce_mobile/lib/models/notifikacija.dart
import 'package:json_annotation/json_annotation.dart';

part 'notifikacija.g.dart';

@JsonSerializable()
class Notifikacija {
  final int? id;
  final int? korisnikId;
  final String? naslov;
  final String? poruka;
  final String? tip;
  final bool? procitano;
  final DateTime? createdAt;
  final int? terminId;

  Notifikacija({
    this.id,
    this.korisnikId,
    this.naslov,
    this.poruka,
    this.tip,
    this.procitano,
    this.createdAt,
    this.terminId,
  });

  factory Notifikacija.fromJson(Map<String, dynamic> json) =>
      _$NotifikacijaFromJson(json);

  Map<String, dynamic> toJson() => _$NotifikacijaToJson(this);
}
